scala-package/core/src/main/scala/ml/dmlc/mxnet/module/DataParallelExecutorGroup.scala

```diff
@@ -0,0 +1,695 @@
+      require(gradReq.size == argNames.size)
+      this.gradReqs = gradReq.toMap
+      this
+    }
+
+    def build(): DataParallelExecutorGroup = {
+      new DataParallelExecutorGroup(
+        symbol, contexts, workLoadList, dataShapes, labelShapes, paramNames, forTraining,
+        inputsNeedGrad, sharedGroup, inputTypes, fixedParamNames, this.gradReqs)
+    }
+  }
+}
+
+/**
+ * DataParallelExecutorGroup is a group of executors that lives on a group of devices.
+ * This is a helper class used to implement data parallelism. Each mini-batch will
+ * be split and run on the devices.
+ * @param symbol The common symbolic computation graph for all executors.
+ * @param contexts A list of contexts.
+ * @param workLoadList If not `None`, could be a list of numbers that
+ *                     specify the workload to be assigned to different context.
+ *                     Larger number indicate heavier workload.
+ * @param dataShapes Should be a list of (name, shape) tuples, for the shapes of data.
+ *                   Note the order is important and should be the same as the order that
+ *                   the `DataIter` provide the data.
+ * @param labelShapes Should be a list of (name, shape) tuples, for the shapes of label.
+ *                    Note the order is important and should be the same as the order that
+ *                    the `DataIter` provide the label.
+ * @param paramNames A list of strings, indicating the names of parameters
+ *                   (e.g. weights, filters, etc.) in the computation graph.
+ * @param forTraining Indicate whether the executors should be bind for training.
+ *                    When not doing training, the memory for gradients will not be allocated.
+ * @param inputsNeedGrad Indicate whether the gradients for the input data should be computed.
+ *                       This is currently not used.
+ *                       It will be useful for implementing composition of modules.
+ * @param sharedGroup Default is `None`. This is used in bucketing. When not `None`,
+ *                    it should be a executor group corresponding to a different bucket.
+ *                    In other words, it will correspond to a different symbol but
+ *                    with the same set of parameters (e.g. unrolled RNNs with different lengths).
+ *                    In this case, many memory will be shared.
+ * @param inputTypes Default is `None`. When not `None`,
+ *                   can be used to specify the data type for each of the data/label inputs.
+ * @param fixedParamNames Indicate parameters to be fixed during training.
+ *                        Parameters in this list will not allocate space for gradient,
+ *                        nor do gradient calculation.
+ * @param gradReq Requirement for gradient accumulation. Can be 'write', 'add', or 'null',
+ *                be specified for each argument.
+ */
+class DataParallelExecutorGroup private[mxnet](
```

Comment:
maybe private to the package? it has a package-private constructor and I didn't see a reason to make it visiable in scaladoc?